tests/test_modeling_pretrained_lm_generation.py

```diff
@@ -0,0 +1,98 @@
+# coding=utf-8
```

Comment:
I created this new test file to test some hard coded generated text for a given pytorch seed on pretrained models. 
Not sure whether this structure: 

1. Condensing the tests for all LMHeadModels in one file &
2. using a utils_generation.py file to collect the hard-coded input & output for the different models 

is the right approach.
Happy about feedback @LysandreJik  & @sshleifer !

Reply:
The issue with this approach is the location of the code we need to update on each model addition/changes. The way the tests are structured is so that creating a new test class inherits from a mixin, which will run all the common tests on the different models defined in the child class.

This means that adding a new model would only require editing that model's file, whereas in this situation, you would have to edit this `test_modeling_pretrained_lm_generation.py` file to add new models and test their `generate` method.

What I like about this is that the code to be added on each model addition is as little as it gets, which is what we're trying to do with the Mixins. 

What I would do would be to add the `generate` tests to `modeling_common_tests`, and define an empty list of models to be tested by this method at the Mixin level, which would be overridden by the test classes which have an actual model to be tested.

Happy to discuss if you disagree/want more details @sshleifer @patrickvonplaten

Reply:
I like @LysandreJik 's approach, but would definitely keep it in a subsequent, separate PR :).

Reply:
Got it! Will add these test then at a later stage and focus on the "randomly initialized lm model" tests.

Reply:
Moved implementation to PR #2909